### ISSUE TITLE
RemovedInDjango40Warning: The providing_args argument is deprecated

### ISSUE THREAD

**Issue by nip3o** [wagtail/wagtail#6924]
### Issue Summary
Since [Django 3.1](https://docs.djangoproject.com/en/3.1/releases/3.1/) the `providing_args` argument to `Signal` is deprecated, which results in the following deprecation log message:

```
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:4: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  page_published = Signal(providing_args=['instance', 'revision'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:5: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  page_unpublished = Signal(providing_args=['instance'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:6: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  pre_page_move = Signal(providing_args=['instance', 'parent_page_before', 'parent_page_after', 'url_path_before', 'url_path_after'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:7: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  post_page_move = Signal(providing_args=['instance', 'parent_page_before', 'parent_page_after', 'url_path_before', 'url_path_after'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:9: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  workflow_approved = Signal(providing_args=['instance', 'user'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:10: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  workflow_rejected = Signal(providing_args=['instance', 'user'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:11: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  workflow_cancelled = Signal(providing_args=['instance', 'user'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:12: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  workflow_submitted = Signal(providing_args=['instance', 'user'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:14: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  task_approved = Signal(providing_args=['instance', 'user'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:15: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  task_rejected = Signal(providing_args=['instance', 'user'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:16: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  task_submitted = Signal(providing_args=['instance', 'user'])
/usr/local/lib/python3.7/dist-packages/wagtail/core/signals.py:17: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  task_cancelled = Signal(providing_args=['instance' 'user'])
/usr/local/lib/python3.7/dist-packages/wagtail/documents/models.py:205: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  document_served = Signal(providing_args=['request'])
/usr/local/lib/python3.7/dist-packages/wagtail/admin/signals.py:4: RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
  init_new_page = Signal(providing_args=['page', 'parent'])
```
It seems like this is still the case [in master](https://github.com/wagtail/wagtail/blob/main/wagtail/core/signals.py) and I could not find an existing issue for this.

### Technical details

* Python version: 3.7.9
* Django version: 3.1.7
* Wagtail version: 2.11.1

**Comment by gasman** [comment]
Thanks for reporting, @nip3o!

As I understand it, Django's recommendation for extensions like Wagtail is that they shouldn't attempt to address compatibility with Django 4.x until they've dropped support for the 2.x branch - we'll most likely do that after Django 3.2 LTS is released, scheduled for next month.

**Comment by thibaudcolas** [comment]
Fixed in #f8299885aa916becb9109c881f3ab28408af794e.
